### PR TITLE
feat(fw): Call `evmone-eofparse` on EOF tests filling

### PR DESCRIPTION
**Pull request by marioevz**
## 🗒️ Description
Adds EOF exception checking using evmone-eofparse tool. Prints a nice message now if expected exception does not match returned by evmone or if valid eof test fails on evmone and so on.

## ✅ Checklist
- [x] All: Set appropriate labels for the changes.
- [x] All: Considered squashing commits to improve commit history.
- [x] All: Added an entry to [CHANGELOG.md](/ethereum/execution-spec-tests/blob/main/docs/CHANGELOG.md).
- [x] All: Considered updating the online docs in the [./docs/](/ethereum/execution-spec-tests/blob/main/docs/) directory.
- [x] Tests: All converted JSON/YML tests from [ethereum/tests](/ethereum/tests) have been added to [converted-ethereum-tests.txt](/ethereum/execution-spec-tests/blob/main/converted-ethereum-tests.txt).
- [x] Tests: Included the type and version of evm t8n tool used to locally execute test cases:  e.g., ref with commit hash or geth 1.13.1-stable-3f40e65.
- [x] Tests: Ran `mkdocs serve` locally and verified the auto-generated docs for new tests in the [Test Case Reference](https://ethereum.github.io/execution-spec-tests/main/tests/) are correctly formatted.

### CODE REVIEW

**Comment by winsvega** [comment]
this PR consumes https://github.com/ethereum/execution-spec-tests/pull/516  that introduced  this feature example use

File: src/ethereum_test_tools/spec/eof/eof_test.py [hunk 1]
```diff
@@ -48,6 +84,51 @@ def make_eof_test_fixture(
...
+        for _, vector in fixture.vectors.items():
+            expectedResult = vector.results.get(str(fork))
+            result = eof_parse.run(input=str(vector.code))
+            self.verify_result(result, expectedResult, vector.code)
+
+        return fixture
+
+    def verify_result(self, result: CompletedProcess, expectedResult: Result, code: Bytes):
+        parser = EvmoneExceptionParser()
+        res_error = result.stdout.replace("\n", "")
```
Comment (winsvega) [inline src/ethereum_test_tools/spec/eof/eof_test.py:97]: currently eof tool does not return an error code, so we analyze the stdout

File: src/ethereum_test_tools/spec/eof/eof_test.py [hunk 2]
```diff
@@ -48,6 +84,51 @@ def make_eof_test_fixture(
...
+                    return
+
+                msg = "EOF code expected to fail with a different exception, than reported:"
+                formatted_message = (
+                    f"{msg} \n"
+                    f"{code} \n"
+                    f"Expected: {expRes} ({parser.parse_exception(expRes)}) \n"
+                    f"     Got: {parser.rev_parse_exception(res_error)} ({res_error})"
+                )
+                raise Exception(formatted_message)
```
Comment (winsvega) [inline src/ethereum_test_tools/spec/eof/eof_test.py:131]: this exception verification feature can be done in the same manner for state tests/bc tests with different tools. 
would be nice to see it in consume.  and also when generating the tests via t8n, the exception string from t8n can be verified against our exception maps

**Comment by shemnon** [comment]
It would be nice if this could be made generic in a manner like TransitionTool is made generic, I expect each client to expose their own tool.

Also, I would like a format that accepts required arguments.  Besu's equivelant is `evmtool code-validation` and that won't pass the binary string matching task ask it is a binary plus an argument.

Combine this with the fork being validated possibly being another argument.

**Comment by marioevz** [comment]
> It would be nice if this could be made generic in a manner like TransitionTool is made generic, I expect each client to expose their own tool.
> 
> Also, I would like a format that accepts required arguments. Besu's equivelant is `evmtool code-validation` and that won't pass the binary string matching task ask it is a binary plus an argument.
> 
> Combine this with the fork being validated possibly being another argument.

I think @danceratopz might know a bit better than I do how to implement something along these lines.

We might need to create another package like we currently have `evm_transition_tool`, maybe a `eof_tool` with all different flavors of EOF tools implemented.

We might also need to implement new parameters to configure it in the `pytest_plugins/test_filler`.

My main ask when we implement this is that the normal test filling process does not fail if there no eof tool installed, only fails if we are trying to fill an EOF test.

File: src/ethereum_test_tools/spec/eof/eof_test.py (hunk 1 above)
Comment (danceratopz) [reply src/ethereum_test_tools/spec/eof/eof_test.py:113]: would be nice to have error codes!

**Review by winsvega — APPROVED** [review]

**Comment by winsvega** [comment]
we stick to evmone-eofparse for verification for now till later when we refactor tooling capture

**Review by danceratopz — COMMENTED** [review]
Very nice.

Really like how/where `eofparse` is called, I had a mental block on this one and wanted to add it to the `FixtureCollector` - it's much more logical in the fixture generator for this use case!

Love the exceptions raised by `verify_fixtures()` - really clear.

After aligning with @winsvega, I've refactored the code a bit in #538, which addresses the small concerns below. Please let me know what you think.

From the above comment from @shemnon:
> It would be nice if this could be made generic in a manner like TransitionTool is made generic, I expect each client to expose their own tool.

We will definitely do this! But will add this work in a separate PR as it requires quite a bit of code organization. Don't want to block this helpful addition though, so we should get this merged first.

**Comment by danceratopz** [comment]
> My main ask when we implement this is that the normal test filling process does not fail if there no eof tool installed, only fails if we are trying to fill an EOF test.

@marioevz, Regarding this, we warn at the end of the test session if `evmone-eofparse` was not found:
image

File: src/ethereum_test_tools/spec/eof/eof_test.py (hunk 2 above)
Comment (danceratopz) [reply src/ethereum_test_tools/spec/eof/eof_test.py]: Yup, definitely!

**Pull request merged**